icontrol/session.py

```diff
@@ -86,11 +86,21 @@ def _validate_icruri(base_uri):
     scheme, netloc, path, _, _ = urlsplit(base_uri)
     if scheme != 'https':
         raise InvalidScheme(scheme)
-    if not path.startswith('/mgmt/tm/'):
-        error_message = "The path must start with '/mgmt/tm/'!!  But it's:" +\
-            " '%s'" % path[:10]
+
+    if path.startswith('/mgmt/tm/'):
+        # Most of the time this is BIG-IP
+        sub_path = path[9:]
+    elif path.startswith('/mgmt/cm/'):
+        # This can also be in iWorkflow or BIG-IQ
+        sub_path = path[9:]
+    elif path.startswith('/mgmt/shared/'):
+        # This can be iWorkflow or BIG-IQ
+        sub_path = path[13:]
+    else:
+        error_message = "The path must start with '/mgmt/tm/'!  But it's:" + \
```

Comment:
Error message still isn't correct because the path can start with things other than `/mgmt/tm/`.